### PR TITLE
Change password using Admin user 

### LINKED ISSUE #4588: Change Password API for Admin Users
Fallout from #4538 

We need to implement admin support for changing a user's password once support for this has been released by ROS. It has been merged but is awaiting a release: https://github.com/realm/realm-object-server/pull/1061

API:

```
// Should only work for `admin` users
user.changePasswordAsync("userID", "newPassword", new Callback() {
       @Override
       public void onSuccess(SyncUser user) {
         // This standard callback might be a bit weird since we would return the admin user instead of 
              the user that got his password changed, 
       }
       @Override
       public void onError(ObjectServerError error) {
         // Something went wrong
       }
});

public interface RequestCallback<T> {
  onSuccess(T result);
  onError(ObjectServerError);
}

new RequestCallback<Void>() {
  public void onSuccess(Void result) { ... }
}


```
Some alternatives:
- New `AdminCallback` with `onSuccess(String userId)` method instead. Might be handy if we start adding multiple admin API's.
- New `RequestCallback()` with `onSuccess()`. The PermissionWrapper could probably use this as well.

- Just re-use existing `Callback` and accept it being slightly weird.

TODO
- [ ] Figure out exactly what the callback should be
- [ ] Implementation 
- [ ] Unit tests
- [ ] Integration tests.

### PR DESCRIPTION
**Pull request by nhachicha**
fixes #4588

### CODE REVIEW

**Comment by nhachicha** [comment]
@realm/java

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 * [ObjectServer] Added two options to `SyncConfiguration` to provide a trusted root CA `trustedRootCA` and to disable SSL validation `disableSSLVerification` (#4371).
```
Comment (cmelchior) [inline CHANGELOG.md]: Whitespace

File: realm/realm-library/src/androidTestObjectServer/java/io/realm/SyncUserTests.java [hunk 2]
```diff
@@ -305,11 +313,47 @@ public void onError(ObjectServerError error) {
     }
 
     @Test
+    public void changePassword_admin_Async_nonLooperThreadThrows() {
+        SyncUser user = createTestUser();
+
+        thrown.expect(IllegalStateException.class);
+        user.changePasswordAsync("old", "new", new SyncUser.Callback() {
```
Comment (cmelchior) [inline realm/realm-library/src/androidTestObjectServer/java/io/realm/SyncUserTests.java]: "old" is the user identifier, right? I would rename that

File: realm/realm-library/src/objectServer/java/io/realm/SyncUser.java [hunk 3]
```diff
@@ -334,6 +331,44 @@ public void changePassword(String newPassword) throws ObjectServerError {
...
+     * This user needs admin privilege in order to change someone else's password.
+     *
+     * <b>WARNING:</b> Changing a users password using an authentication server that doesn't use HTTPS is a major
+     * security flaw, and should only be done while testing.
+     *
+     * @param userID identity ({@link #getIdentity()}) of the user we want to change the password for.
+     * @param newPassword the user's new password.
+     * @throws ObjectServerError if the password could not be changed.
+     */
+    public void changePassword(final String userID, final String newPassword) throws ObjectServerError {
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/SyncUser.java]: How strict are we with camelCase, should it be `UserId` ?

File: realm/realm-library/src/objectServer/java/io/realm/SyncUser.java [hunk 4]
```diff
@@ -334,6 +331,44 @@ public void changePassword(String newPassword) throws ObjectServerError {
...
+        if (newPassword == null) {
+            throw new IllegalArgumentException("Not-null 'newPassword' required.");
+        }
+
+        if (Util.isEmptyString(userID)) {
+            throw new IllegalArgumentException("None empty 'userID' required.");
+        }
+
+        if (userID.equals(getIdentity())) { // user want's to change his password
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/SyncUser.java]: change his own password

File: realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/AuthTests.java [hunk 5]
```diff
@@ -195,6 +194,72 @@ public void changePassword() {
     }
 
     @Test
+    public void changePassword_using_admin() {
+        String username = UUID.randomUUID().toString();
+        String originalPassword = "password";
+        SyncCredentials credentials = SyncCredentials.usernamePassword(username, originalPassword, true);
+        SyncUser userOld = SyncUser.login(credentials, Constants.AUTH_URL);
+        assertTrue(userOld.isValid());
+
```
Comment (cmelchior) [inline realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/AuthTests.java]: Whitespace

File: realm/realm-library/src/objectServer/java/io/realm/SyncUser.java [hunk 6]
```diff
@@ -334,6 +331,44 @@ public void changePassword(String newPassword) throws ObjectServerError {
     }
 
     /**
+     * Changes another user's password. This is done synchronously and involves the network, so calling this method on the
+     * Android UI thread will always crash.
+     * <p>
+     * This user needs admin privilege in order to change someone else's password.
+     *
+     * <b>WARNING:</b> Changing a users password using an authentication server that doesn't use HTTPS is a major
```
Comment (beeender) [inline realm/realm-library/src/objectServer/java/io/realm/SyncUser.java]: `users` -> `user's` `using` -> `through`
hanging a user`s password through an authentication server that doesn't use HTTPS is a major

**Pull request merged**